api/v1beta1/module_webhook_test.go

```diff
@@ -0,0 +1,263 @@
+					Container: ModuleLoaderContainerSpec{
+						KernelMappings: []KernelMapping{
+							{Regexp: "", ContainerImage: "image-url"},
+						},
+					},
+				},
+			},
+		}
+
+		mod2 := &Module{
+			Spec: ModuleSpec{
+				ModuleLoader: ModuleLoaderSpec{
+					Container: ModuleLoaderContainerSpec{
+						ContainerImage: "image-url",
+						KernelMappings: []KernelMapping{
+							{Regexp: "valid-regex"},
+						},
+					},
+				},
+			},
+		}
+
+		Expect(mod1.validateKernelMappingRegex()).ToNot(HaveOccurred())
+		Expect(mod2.validateKernelMappingRegex()).ToNot(HaveOccurred())
+	})
+
+	It("should fail when an invalid regex is found", func() {
+		mod := &Module{
+			Spec: ModuleSpec{
+				ModuleLoader: ModuleLoaderSpec{
+					Container: ModuleLoaderContainerSpec{
+						KernelMappings: []KernelMapping{
+							{Regexp: "*-invalid-regex"},
+						},
+					},
+				},
+			},
+		}
+
+		e := mod.ValidateCreate()
+		Expect(e).To(HaveOccurred())
+		Expect(e.Error()).To(ContainSubstring("invalid regexp"))
+	})
+
+	It("should fail when a kernel-mapping has invalid containerName", func() {
+		mod := &Module{
+			Spec: ModuleSpec{
+				ModuleLoader: ModuleLoaderSpec{
+					Container: ModuleLoaderContainerSpec{
+						Modprobe: ModprobeSpec{
+							ModuleName: "mod-name",
+						},
+						KernelMappings: []KernelMapping{
+							{Regexp: "valid-regex"},
+						},
+					},
+				},
+			},
+		}
+
+		e := mod.ValidateCreate()
+		Expect(e).To(HaveOccurred())
+		Expect(e.Error()).To(
+			ContainSubstring(
+				"missing spec.moduleLoader.container.kernelMappings[].containerImage",
+			),
+		)
+	})
+})
+
+var _ = Describe("validateModuleName", func() {
+	It("should fail when moduleName is missing", func() {
+		mod := &Module{}
+
+		e := mod.validateModuleName()
+		Expect(e).To(HaveOccurred())
+		Expect(e.Error()).To(ContainSubstring("missing spec.moduleLoader.container.modprobe.moduleName"))
+	})
+
+	It("should fail when rawArgs.load is empty", func() {
+		mod := &Module{
+			Spec: ModuleSpec{
+				ModuleLoader: ModuleLoaderSpec{
+					Container: ModuleLoaderContainerSpec{
+						Modprobe: ModprobeSpec{
+							RawArgs: &ModprobeArgs{
+								Load: nil,
+							},
+						},
+					},
+				},
+			},
+		}
+
+		e := mod.validateModuleName()
+		Expect(e).To(HaveOccurred())
+		Expect(e.Error()).To(ContainSubstring("missing spec.moduleLoader.container.modprobe.rawArgs.load"))
+	})
+
+	It("should fail when rawArgs.unload is empty", func() {
+		mod := &Module{
+			Spec: ModuleSpec{
+				ModuleLoader: ModuleLoaderSpec{
+					Container: ModuleLoaderContainerSpec{
+						Modprobe: ModprobeSpec{
+							RawArgs: &ModprobeArgs{
+								Load:   []string{"arg"},
+								Unload: nil,
+							},
+						},
+					},
+				},
+			},
+		}
+
+		e := mod.validateModuleName()
+		Expect(e).To(HaveOccurred())
+		Expect(e.Error()).To(ContainSubstring("missing spec.moduleLoader.container.modprobe.rawArgs.unload"))
+	})
+
+	It("should pass when rawArgs has load and unload values", func() {
```

Comment:
We also need a test to validate that it passes when `spec.moduleLoader.container.modporbe.name` is set (with no rawargs).

Reply:
Yup. Sorry, this is hard to follow when non-interface helper-function are used :)
If it is cover in another test then no need to duplicate. Sorry for the noise.